spec/jobs/email_privacy_policy_update_job_spec.rb

```diff
@@ -4,25 +4,28 @@
   describe "perform" do
     context "signature validated, without matching privacy_notification" do
       let(:petition) { FactoryBot.create(:petition) }
+      let(:time) { 1.year.ago }
 
       let!(:signature) do
         FactoryBot.create(:validated_signature, petition: petition)
       end
 
+      let(:subject) do
+        described_class.perform_now(petition: petition, time: time)
+      end
+
       it "sends an email" do
-        expect {
-          described_class.perform_now(petition)
-        }.to change {
-          enqueued_jobs.count
-        }.by(1)
+        expect { subject }.to change { enqueued_jobs.count }.by(1)
       end
 
       it "creates a privacy_notification record" do
-        expect {
-          described_class.perform_now(petition)
-        }.to change {
-          PrivacyNotification.count
-        }.by(1)
+        expect { subject }.to change { PrivacyNotification.count }.by(1)
+      end
+
+      it "sets ignore_petitions_before on privacy_notification" do
+        time = 1.year.ago
+        described_class.perform_now(petition: petition, time: time)
+        expect(PrivacyNotification.last.ignore_petitions_before).to eq(time)
```

Comment:
Change this to: 

``` ruby
expect(PrivacyNotification.last.ignore_petitions_before).to be_within(1.second).of time
```

The PostgreSQL timestamp resolution is to 6 decimal places but Linux's clock returns nanoseconds (MacOS returns nanoseconds but they're always 000 so it passes).

Reply:
Aha...thank you. I was tearing my hair out 😅